collab-document/src/blocks/blockv2.rs

```diff
@@ -0,0 +1,136 @@
+use std::collections::HashMap;
+
+use collab::preclude::{Map, MapRefWrapper, ReadTxn, TransactionMut};
+use serde::{
+  ser::{SerializeMap, SerializeStruct},
+  Deserialize, Serialize, Serializer,
+};
+use serde_json::Value;
+
+use crate::{document::Document, error::DocumentError};
+
+const BLOCK_ID: &str = "id";
+const BLOCK_TYPE: &str = "ty";
+const BLOCK_PARENT: &str = "parent";
+const BLOCK_CHILDREN: &str = "children";
+const BLOCK_DATA: &str = "data";
+
+#[derive(Deserialize, Debug)]
+pub struct BlockV2 {
+  pub id: String,
+  pub ty: String,
+  pub parent: String,
+  pub children: String,
+  pub data: HashMap<String, Value>,
+}
+
+impl Serialize for BlockV2 {
+  fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+  where
+    S: Serializer,
+  {
+    let mut state = serializer.serialize_struct("Block", 5)?;
+    state.serialize_field(BLOCK_ID, &self.id)?;
+    state.serialize_field(BLOCK_TYPE, &self.ty)?;
+    state.serialize_field(BLOCK_PARENT, &self.parent)?;
+    state.serialize_field(BLOCK_CHILDREN, &self.children)?;
+    state.serialize_field(BLOCK_DATA, &self.data)?;
+    state.end()
+  }
+}
+
+/// {
+///   "block_map": root,
+/// }
+pub struct BlockMapV2 {
+  root: MapRefWrapper,
+}
+
+impl Serialize for BlockMapV2 {
```

Comment:
Maybe you can try to create a helper serde struct.  For example: the `DatabaseSerde`(in the main branch). Then you don't have to handle the serde stuff. 

Implement the serde by hand made in these conditions:
1.  provide some custom transformation
2.  migration